src/Invoice_GenUI/Invoice_GenUI/Models/Services/ClientService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Invoice_GenUI.Models.Services
+{
+    public interface IClientService
+    {
+        Task<List<ClientNameModel>> GetClientNames();
+        Task<List<CreateClientModel>> GetClientDetails();
+        Task<SingleClientModel> GetSingleClientDetails(int index);
+        Task<bool> PutClient(CreateClientModel newClient);
+    }
+    public partial class ClientService : IClientService
+    {
+        public async Task<List<ClientNameModel>> GetClientNames()
+        {
+            using (var client = new HttpClient())
```

Comment:
:seedling: now that we're using instances of the `HttpClient` class in multiple places in this class, I'll create a task for clearing this up and will walk you through it.